### DISCUSSION TITLE
404 Not Found. Unable to render this definition The provided definition does not specify a valid version field.

### DISCUSSION THREAD

**Discussion by Ivareh** [fastapi/fastapi#11920] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
"""
My repo is https://github.com/Path-of-Modifiers/pathofmodifiersapp/tree/325-fix-issue-404-not-found-in-backend-api-production . Maybe you can reproduce the error here, but I will provide more information if needed. 

main.py:


"""

from __future__ import annotations
from fastapi import FastAPI
from starlette.middleware.cors import CORSMiddleware

from app.api.api_v1.api import api_router

from app.core.config import settings


app = FastAPI(
    title=settings.PROJECT_NAME,
    openapi_url=f"{settings.API_V1_STR}/openapi.json",
    # generate_unique_id_function=custom_generate_unique_id,
)


@app.get("/")
async def read_main():
    return {"message": "Welcome to Path of Modifiers API!"}


# Set all CORS enabled origins
if settings.BACKEND_CORS_ORIGINS:
    app.add_middleware(
        CORSMiddleware,
        allow_origins=[
            str(origin).strip("/") for origin in settings.BACKEND_CORS_ORIGINS
        ],
        allow_credentials=True,
        allow_methods=["*"],
        allow_headers=["*"],
    )


app.include_router(api_router, prefix=settings.API_V1_STR)
```

### Description

Everything seems to work fine in localhost with 200 responses and /docs displays the document correctly. 

But when I run FastAPI application in production with a custom domain on a remote server, in this case domain pathofmodifiers.com, I run into a problem that the API returns 404 Not Found on all requests. There are separate error 404 page for frontend in any non-related API urls, which works correctly. When I go to /docs, I get the error:

```
Unable to render this definition
The provided definition does not specify a valid version field.

Please indicate a valid Swagger or OpenAPI version field. Supported version fields are swagger: "2.0" and those that match openapi: 3.x.y (for example, openapi: 3.1.0).
```

Stack:
- FastAPI backend
- SQLAlchemy ORM
- Pydantic 
- PostgreSQL database
- React Typescript Vue
- Traefik reverse proxy

When I replace my backend_api with https://github.com/tiangolo/full-stack-fastapi-template/tree/master's backend module and build it in production with the custom domain, the application seems to work fine. So there is most likely something in my backend module that's wrong.

Things I have tried:
- Remove all endpoints and run the application
- Not include router
- Change the open_api.json url
- Remove backend_prestart scripts with alembic migrations
- https://github.com/fastapi/fastapi/issues/4986

### Operating System

Linux

### Operating System Details

Ubuntu 22 Remote server

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

3.11

### Additional Context

![Screenshot from 2024-07-30 10-14-38](https://github.com/user-attachments/assets/7e7209d1-f36b-4891-b755-a77f218125c8)
![Screenshot from 2024-07-30 10-15-02](https://github.com/user-attachments/assets/611a8319-dd28-4fdf-96ae-b8292f3b8446)

**Comment by gabrielcarvajalfigueroa** [comment]
Hi @Ivareh ,

It depends on your reverse proxy configuration, I assume that you are serving FrontEnd and Backend through port 443 and that your backend serves the responses if the url contains ```/api```. If that's the case your `main.py` file should look something like this

```py
from fastapi import FastAPI

app = FastAPI(root_path='/api')

app.include_router(api_router, prefix=settings.API_V1_STR)
```

But the main thing is that the `root_path` that you pass must match the config that you have in your reverse proxy. It might happen that the docs will work and the endpoints will stop working if that happens just check what value you pass here: `prefix=settings.API_V1_STR` and make it fit to the reverse proxy configuration.

Hope this helps

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
